lib/prototypes/abstract.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const { formatters, checks, harmonize } = require('../util.js');
+
+const AbstractType = {
+  formatters: {},
+  checks: {},
+  check(value, path, root) {
+    try {
+      const typeErr = harmonize(this.checkType(value, path, root));
+      if (typeErr) return typeErr;
+      if (this.validate) {
+        const validErr = harmonize(this.validate(value, path, root));
+        if (validErr) return validErr;
+      }
+      for (const [name, subCheck] of Object.entries(this.checks)) {
+        if (this[name]) subCheck(value, path, this);
```

Comment:
What if one of the `subCheck` wants to return an error, shouldn't we collect these into an array and return it?

Reply:
I did not use harmonize for subCheck. And I do not expect to run all checks, so I will add this option, cause we have similar behavior in Schema. Thx

Reply:
Changed to an array

Reply:
Here we need to check if there is subCheck rule in type definition. Can not be changed.

Reply:
Answered